main.go

```diff
@@ -78,6 +78,31 @@ func main() {
 		regArr = append(regArr, re)
 	}
 
+	if len(flag.Args()) == 0 {
+		getBuf(os.Stdin, 0, &counter, nInvertParameter, quietParameter, maxCountParameter, regArr)
+	} else {
+		for i := range flag.Args() {
+			file, fileErr = os.Open(flag.Args()[i])
+			if fileErr != nil {
+				fmt.Fprintln(os.Stderr, fileErr)
+				os.Exit(2)
+			}
+
+			getBuf(file, i, &counter, nInvertParameter, quietParameter, maxCountParameter, regArr)
+			file.Close()
+		}
+	}
+
+	if counter <= 0 {
+		os.Exit(1)
+	}
+}
+
+func getBuf(file *os.File, u int, counter *int, nInvertParameter *bool, quietParameter *bool, maxCountParameter *int, regArr []*regexp.Regexp) {
```

Comment:
... which allowed you to shorten this one a bit.

Reply:
i did make it global becuase i need in the main and the getBuf function but it still needs to be the same var

Reply:
Yes and now you can shorten the function signature a bit.